src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -694,6 +696,121 @@ def _private_ec_key_to_jwk(ec_key, jwk):
         key_obj.t = byok_data
 
     return client.import_key(vault_base_url, key_name, key_obj, protection == 'hsm', key_attrs, tags)
+
+
+def download_key(client, file_path, vault_base_url=None, key_name=None, key_version='',
+                 encoding=None, identifier=None):  # pylint: disable=unused-argument
+    """ Download a key from a KeyVault. """
+    if os.path.isfile(file_path) or os.path.isdir(file_path):
+        raise CLIError("File or directory named '{}' already exists.".format(file_path))
+
+    def _bytes_to_int(b):
```

Comment:
it will be great to add comments on where the code comes from for future maintainance

Reply:
@yungezz Good idea, I will add some comments for the complicated code parts.